crates/claims/core/src/verification/proof.rs

```diff
@@ -108,7 +108,7 @@ impl From<std::convert::Infallible> for ProofValidationError {
 pub type ProofValidity = Result<(), InvalidProof>;
 
 /// Proof that can be validated against `T` claims with a verifier of type `V`.
-pub trait ValidateProof<T, E, V> {
+pub trait ValidateProof<V, T> {
```

Comment:
Is there some guidance from the specs on the distinction between verification and validation? IMO validation is for "static checks" (e.g. making sure it's well formatted) and verification is a superset of validation. But then you can say that a signature is "valid", so I don't know.

Reply:
As far as I understand validation is checking that the claims themselves are compatibles with the verifier. That can mean checking the claims are not expired, the issuer is supported, the verifier is part of the claims audience, etc. Well formlessness checks that should be handled by the type system as much as possible, but more complex well-formlessness properties could also be checked at runtime during validation.

But as you said, the specs also often use the term "valid" for signatures. That's why to conciliate the two notions I've introduce the two traits `ValidateClaims` for the former understanding regarding claims, and `ValidateProof` for the later regarding signatures. Then "verifying" consists in validating the claims AND the signature.